exasol/bucketfs/_buckets.py

```diff
@@ -223,6 +232,82 @@ def download(self, path: str, chunk_size: int = 8192) -> Iterable[ByteString]:
             yield from response.iter_content(chunk_size=chunk_size)
 
 
+class SaaSBucket:
+
+    def __init__(self, url: str, account_id: str, database_id: str, pat: str) -> None:
+        self._url = url
+        self._account_id = account_id
+        self.database_id = database_id
+        self._pat = pat
+
+    def name(self) -> str:
+        # TODO: Find out the name of the bucket in SaaS
+        return 'default'
+
+    def files(self) -> Iterable[str]:
+        """To be provided"""
+        raise NotImplementedError()
+
+    def delete(self, path: str) -> None:
+        """To be provided"""
+        raise NotImplementedError()
+
+    def upload(self, path: str, data: ByteString | BinaryIO) -> None:
+        """To be provided"""
+        raise NotImplementedError()
+
+    def download(self, path: str, chunk_size: int = 8192) -> Iterable[ByteString]:
+        """To be provided"""
+        raise NotImplementedError()
+
+
+class MountedBucket:
+    """
+    Implementation of the Bucket interface backed by a normal file system in read-only mode.
+    The targeted use case is the read-only access to the BucketFS files from a UDF.
+
+    Q. What exception should be raised when the user attempts to download non-existing file?
+    A.
+
+    Q. What exception should be raised if the user tries to delete or upload a file?
+    A.
+    """
+
+    def __init__(self,
+                 service_name: str,
+                 bucket_name: str):
+        self._name = bucket_name
+        self.root = Path(service_name) / bucket_name
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @property
+    def files(self) -> list[str]:
+        root_length = len(str(self.root))
+        if self.root != self.root.root:
+            root_length += 1
+        return [str(pth)[root_length:] for pth in self.root.rglob('*.*')]
+
+    def delete(self, path: str) -> None:
+        raise PermissionError('File deletion is not allowed.')
+
+    def upload(self, path: str, data: ByteString | BinaryIO) -> None:
+        raise PermissionError('Uploading a file is not allowed.')
+
+    def download(self, path: str, chunk_size: int) -> Iterable[ByteString]:
+        full_path = self.root / path
```

Comment:
Need to think if the `path` contains `service_name/bucket_name` or not.

Reply:
see comment above.